src/modelinstance.hpp

```diff
@@ -216,31 +216,7 @@ class ModelInstance {
          */
     Status loadOVCNNNetworkUsingCustomLoader();
 
-    const Status checkIfShapeValuesNegative(const tensorflow::TensorProto& requestInput);
-
-    virtual const Status validateNumberOfInputs(const tensorflow::serving::PredictRequest* request,
-        const size_t expectedNumberOfInputs);
-
-    const Status validatePrecision(const ovms::TensorInfo& networkInput,
-        const tensorflow::TensorProto& requestInput);
-
-    const Status validateNumberOfShapeDimensions(const ovms::TensorInfo& networkInput,
-        const tensorflow::TensorProto& requestInput);
-
-    const Status validateNumberOfBinaryInputShapeDimensions(const tensorflow::TensorProto& requestInput);
-
-    const bool checkBinaryInputBatchSizeMismatch(const ovms::TensorInfo& networkInput,
-        const tensorflow::TensorProto& requestInput);
-
-    const bool checkBatchSizeMismatch(const ovms::TensorInfo& networkInput,
-        const tensorflow::TensorProto& requestInput);
-
-    const bool checkShapeMismatch(const ovms::TensorInfo& networkInput,
-        const tensorflow::TensorProto& requestInput,
-        const Mode& batchingMode);
-
-    const Status validateTensorContentSize(const ovms::TensorInfo& networkInput,
-        const tensorflow::TensorProto& requestInput);
+    virtual const size_t getExpectedNumberOfInputs(const tensorflow::serving::PredictRequest& request) const;
```

Comment:
Stateful model needs to check what inputs are received from client. Stateful accepts SPECIAL_INPUTS optionally, there might be 0 1 or 2 special inputs and it will be ok.

I think it's bad idea to move stateful logic out of StatefulModelInstance class.